app/models/task.py

```diff
@@ -1,5 +1,27 @@
 from app import db
+from datetime import datetime
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal = db.relationship("Goal", back_populates="tasks")
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.id'), nullable=True)
+
+    def to_dict(self):
+        task_as_dict = {
+            "id": self.id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.completed_at != None
```

Comment:
Great choice to derive the value for `is_complete` from `completed_at`!

I want to share a gentle reminder that we should use `is` or `is not` when comparing objects to `None` in python. Another option could be to use the python `bool` function, what could that look like?